enterprise/api/v1/views/analytics_summary.py

```diff
@@ -0,0 +1,58 @@
+"""
+Views for the `ChatGPTResponse` API endpoint.
+"""
+from edx_rbac.decorators import permission_required
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import generics, status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from django.shortcuts import get_object_or_404
+
+from enterprise.api.utils import (
+    generate_prompt_for_learner_engagement_summary,
+    generate_prompt_for_learner_progress_summary,
+)
+from enterprise.api.v1.serializers import AnalyticsSummarySerializer
+from enterprise.models import ChatGPTResponse, EnterpriseCustomer
+
+
+class AnalyticsSummaryView(generics.GenericAPIView):
```

Comment:
If we `generics.CreateAPIView` then do we still need to set ` http_method_names = ['post']`?

Reply:
Yes, we would still need this.